graylog2-web-interface/src/components/events/events/EventDetails.tsx

```diff
@@ -0,0 +1,150 @@
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React from 'react';
+import { PluginStore } from 'graylog-web-plugin/plugin';
+import lodash from 'lodash';
+
+import useCurrentUser from 'hooks/useCurrentUser';
+import { isPermitted } from 'util/PermissionsMixin';
+import { Link } from 'components/common/router';
+import { Col, Row } from 'components/bootstrap';
+import { Timestamp } from 'components/common';
+import Routes from 'routing/Routes';
+import EventDefinitionPriorityEnum from 'logic/alerts/EventDefinitionPriorityEnum';
+import type { Event, EventDefinitionContext } from 'components/events/events/types';
+
+type Props = {
+  event: Event,
+  eventDefinitionContext: EventDefinitionContext,
+};
+
+const EventDetails = ({ event, eventDefinitionContext }: Props) => {
+  const getConditionPlugin = (type: string): EventDefinitionType => {
+    if (type === undefined) {
+      return null;
+    }
+
+    return PluginStore.exports('eventDefinitionTypes').find((edt) => edt.type === type);
+  };
+
+  const renderEventFields = (eventFields) => {
+    const fieldNames = Object.keys(eventFields);
+
+    return (
+      <ul>
+        {fieldNames.map((fieldName) => {
+          return (
+            <React.Fragment key={fieldName}>
+              <li><b>{fieldName}</b> {eventFields[fieldName]}</li>
+            </React.Fragment>
+          );
+        })}
+      </ul>
+    );
+  };
+
+  const currentUser = useCurrentUser();
+  const plugin = getConditionPlugin(event.event_definition_type);
+
+  const renderLinkToEventDefinition = () => {
+    if (!eventDefinitionContext) {
+      return <em>{event.event_definition_id}</em>;
+    }
+
+    return isPermitted(currentUser.permissions,
+      `eventdefinitions:edit:${eventDefinitionContext.id}`)
+      ? <Link to={Routes.ALERTS.DEFINITIONS.edit(eventDefinitionContext.id)}>{eventDefinitionContext.title}</Link>
+      : eventDefinitionContext.title;
+  };
+
+  const renderReplaySearchLink = () => {
+    const replayInfo = event.replay_info;
+    let rangeType;
+    let range;
+    let streams;
+
+    if (replayInfo.timerange_start && replayInfo.timerange_end) {
+      rangeType = 'absolute';
+      range = { from: `${replayInfo.timerange_start}`, to: `${replayInfo.timerange_end}` };
+    }
+
+    if (replayInfo.streams) {
+      streams = replayInfo.streams;
+    }
+
+    const replaySearchUrl = Routes.search_with_query(replayInfo.query, rangeType, range, streams);
+
+    return <Link to={replaySearchUrl}>Replay Search</Link>;
```

Comment:
We have a "Replay Search" button for dashboards widgets. 
![image](https://user-images.githubusercontent.com/46300478/203964651-1d6dacfb-c4f0-48c3-a46c-cb15263df7a9.png)

It looks like we can reuse the component or at least some logic and styling like the icon, by adjusting it slightly.